presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/TestIcebergMaterializedViews.java

```diff
@@ -0,0 +1,348 @@
+
+        assertUpdate("CREATE TABLE test_materialized_views.base_table4 (_varchar VARCHAR, _bigint BIGINT, _date DATE) WITH (partitioning = ARRAY['_bigint', '_date'])");
+        assertUpdate("INSERT INTO test_materialized_views.base_table4 VALUES ('a', 0, CAST('2019-09-08' AS DATE)), ('a', 1, CAST('2019-09-08' AS DATE)), ('a', 0, CAST('2019-09-09' AS DATE))", 3);
+
+        // A partitioned materialized view with grouping and aggregation
+        assertUpdate("CREATE MATERIALIZED VIEW test_materialized_views.materialized_view_part_stale WITH (partitioning = ARRAY['_date']) as select _date, count(_date) as num_dates from test_materialized_views.base_table3 group by 1");
+        // A non-partitioned join materialized view
+        assertUpdate("CREATE MATERIALIZED VIEW test_materialized_views.materialized_view_join_stale as " +
+                "select t2._bigint, _varchar, t1._date from test_materialized_views.base_table3 t1, test_materialized_views.base_table4 t2 where t1._date = t2._date");
+        // A partitioned join materialized view
+        assertUpdate("CREATE MATERIALIZED VIEW test_materialized_views.materialized_view_join_part_stale WITH (partitioning = ARRAY['_date', '_bigint']) as " +
+                "select t1._bigint, _varchar, t2._date, sum(1) as my_sum from test_materialized_views.base_table3 t1, test_materialized_views.base_table4 t2 where t1._date = t2._date group by 1, 2, 3 order by 1, 2");
+
+        // Materialized views to test SQL features
+        assertUpdate("CREATE MATERIALIZED VIEW test_materialized_views.materialized_view_window WITH (partitioning = ARRAY['_date']) as select _date, " +
+                "sum(_bigint) OVER (partition by _date order by _date) as sum_ints from test_materialized_views.base_table1");
+        assertUpdate("REFRESH MATERIALIZED VIEW test_materialized_views.materialized_view_window", 6);
+        assertUpdate("CREATE MATERIALIZED VIEW test_materialized_views.materialized_view_union WITH (partitioning = ARRAY['_date']) as " +
+                "select _date, count(_date) as num_dates from test_materialized_views.base_table1 group by 1 union " +
+                "select _date, count(_date) as num_dates from test_materialized_views.base_table2 group by 1");
+        assertUpdate("REFRESH MATERIALIZED VIEW test_materialized_views.materialized_view_union", 5);
+        assertUpdate("CREATE MATERIALIZED VIEW test_materialized_views.materialized_view_subquery WITH (partitioning = ARRAY['_date']) as " +
+                "select _date, count(_date) as num_dates from test_materialized_views.base_table1 where _date = (select max(_date) from test_materialized_views.base_table2) group by 1");
+        assertUpdate("REFRESH MATERIALIZED VIEW test_materialized_views.materialized_view_subquery", 1);
+
+        // Materialized view to test 'replace' feature
```

Comment:
We prefer to do the initialization close to the tests, since that makes the tests self contained and thus easier to read. Shared resources are also problematic because multiple tests could modify them (even accidentally), and tests run multithreaded in arbitrary order, leading to hard to debug failures.